chart/deploy.py

```diff
@@ -0,0 +1,160 @@
+}
+
+aws_session = boto3.session.Session()
+
+
+def exec_command(commands, timeout=3):
+  try:
+    output = subprocess.check_output(
+      commands, stderr=subprocess.STDOUT, shell=True, timeout=timeout,
+      universal_newlines=True)
+  except subprocess.CalledProcessError as exc:
+    print('`{}` has been failed with error:'.format(commands), exc.returncode, exc.output)
+    raise
+  print('`{}` has been succeed. output: {}'.format(commands, output))
+
+
+def setup_kubeconfig(aws_region, airflow_type, cluster_name=None):
+  if not cluster_name:
+    cluster_name = CLUSTER_NAME_FORMAT.format(airflow_type=airflow_type, region=SHORTENED_AWS_REGIONS[aws_region])
+  unified_name = '{}.{}.eksctl.io'.format(cluster_name, aws_region)
+  configs = None
+
+  if os.path.isfile(KUBECONFIG_PATH):
+    file = open('/Users/harley.son/.kube/config')
+    configs = yaml.load(file, Loader=yaml.FullLoader)
+
+  if configs:
+    try:
+      if any(d['name'] == unified_name for d in configs['clusters']):
+        print('The configuration for the cluster {} already exists.'.format(cluster_name))
+        return unified_name
+    except:
+      print('The {} file has wrong format.'.format(KUBECONFIG_PATH))
+      raise
+  else:
+    configs = {
+      'apiVersion': 'v1',
+      'kind': 'Config',
+      'clusters': [],
+      'contexts': [],
+      'current-context': '',
+      'preferences': {},
+      'users': [],
+    }
+
+  eks = aws_session.client('eks', region_name=aws_region)
+  cluster_name = CLUSTER_NAME_FORMAT.format(airflow_type=airflow_type, region=SHORTENED_AWS_REGIONS[aws_region])
+
+  cluster = eks.describe_cluster(name=cluster_name)
+  cluster_cert = cluster["cluster"]["certificateAuthority"]["data"]
+  cluster_endpoint = cluster["cluster"]["endpoint"]
+
+  configs['clusters'].append({
+    'cluster': {
+      'server': str(cluster_endpoint),
+      'certificate-authority-data': str(cluster_cert),
+    },
+    'name': unified_name,
+  })
+  configs['contexts'].append({
+    'context': {
+      'cluster': unified_name,
+      'user': unified_name,
+    },
+    'name': unified_name,
+  })
+  configs['users'].append({
+    'name': unified_name,
+    'user': {
+      'exec': {
+        'apiVersion': 'client.authentication.k8s.io/v1alpha1',
+        'command': 'aws-iam-authenticator',
+        "args": ["token", "-i", cluster_name]
+      }
+    }
+  })
+  config_text = yaml.dump(configs, default_flow_style=False)
+  open(KUBECONFIG_PATH, 'w').write(config_text)
+
+  print('Configuration setup has been done for the cluster {}'.format(cluster_name))
+  return unified_name
+
+
+def deploy(region, aws_region, airflow_type, cluster_name, first_release):
+  context_name = setup_kubeconfig(aws_region, airflow_type, cluster_name)
+
+  exec_command(['kubectl config use-context {}'.format(context_name)])
+  exec_command(['helm {} airflow {} --namespace {}'.format(
+    'install' if first_release else 'upgrade', os.path.dirname(os.path.abspath(__file__)), region
```

Comment:
Using `helm upgrade -i ` will install the chart if it is not installed yet, or upgrade it otherwise. Then we don't need the `first_release` tag.

Reply:
Good to know that! Thanks